dfsclient/src/main/java/org/hpccsystems/dfs/client/BinaryRecordReader.java

```diff
@@ -965,7 +965,7 @@ private void trimStringInScratchBuffer(boolean isUnicode, int[] range)
             while (range[1] > range[0])
             {
                 // Need to check against EOS (0x0) in trim fixed len strings correctly
-                int codePoint = this.scratchBuffer[range[1]-2] << 8 | this.scratchBuffer[range[1]-1];
+                int codePoint = (this.scratchBuffer[range[1]-2] & 0xFF) | ((this.scratchBuffer[range[1]-1] & 0xFF) << 8);
```

Comment:
what is the intended net effect of the bitwise and operation w/ the FF mask?

Reply:
Idea here is to combine those two bytes into an unsigned short for the unicode code point. Masking by 0xFF to convert from signed byte to unsigned value in an int, shift upper byte by 8 bits, and then OR the lower and upper ints together to get desired unsigned code point.